lib/index.js

```diff
@@ -114,141 +197,78 @@ const os = require('os'),
             events = {},
             scripts = {},
             items = {},
+            consoles = {},
             scriptStartTime,
             iterationStartTime,
             itemStartTime,
-            requestStartTime,
-            consoles = {},
-
-            _events = flattenEvents(cr.collection),
-            _items = flattenItems(cr.collection.items.members);
+            requestStartTime;
 
         newman.on('beforeIteration', function () {
-            iterationStartTime = Date.now();
+            iterationStartTime = Date.now(); // used on the completion of this iteration
         });
 
         newman.on('iteration', function (err, data) {
             if (err) {
-                !(err instanceof SerialisedError) && (err = new SerialisedError(err, true));
-
-                errors[err.id] = err;
+                errors[err.id] = err instanceof SerialisedError ? err : new SerialisedError(err, true);
             }
 
-            const executionId = uuid();
-
-            executions.push({
-                id: executionId,
-                index: executions.length,
-                branch: 0,
-                type: 'iteration',
-                model: {
-                    $self: ['iterations', null]
-                },
-                cursor: data.cursor,
-                context: {
-                    ref: data.cursor.ref,
-                    httpRequestId: data.cursor.httpRequestId,
-                    scriptId: data.cursor.scriptId,
-                    execution: data.cursor.execution
-                },
-                run: {
-                    $self: ['run', runId]
-                },
-                error: err && { $self: ['errors', err.id] },
-                timings: {
-                    started: iterationStartTime,
-                    completed: Date.now()
-                }
-            });
+            addExecution(err,
+                data,
+                executions,
+                EXECUTION_TYPES.iteration,
```

Comment:
Check if it is possible to infer the execution type from the model type or vice versa. This helps remove one argument from the signature.